central/cluster/datastore/datastore_impl_postgres_test.go

```diff
@@ -31,31 +62,775 @@ func TestClusterDataStoreWithPostgres(t *testing.T) {
+	s.NoError(roleBindingUpsertErr)
+	testImageIntegration := &storage.ImageIntegration{Id: fixtureconsts.AlertFake, ClusterId: clusterId}
+	imageIntegrationId, imageIntegrationAddErr := s.imageIntegrationDatastore.AddImageIntegration(ctx, testImageIntegration)
+	s.NotEmpty(imageIntegrationId)
+	s.NoError(imageIntegrationAddErr)
+	doneSignal := concurrency.NewSignal()
+	clusterRemoveErr := s.clusterDatastore.RemoveCluster(ctx, clusterId, &doneSignal)
+	s.NoError(clusterRemoveErr)
+	s.True(concurrency.WaitWithTimeout(&doneSignal, 10*time.Second))
+	_, deploymentFound, deploymentGetErr := s.deploymentDatastore.GetDeployment(ctx, testDeployment.GetId())
+	s.NoError(deploymentGetErr)
+	s.False(deploymentFound)
+	_, podFound, podGetErr := s.podDatastore.GetPod(ctx, testPod.GetId())
+	s.NoError(podGetErr)
+	s.False(podFound)
+	alert, alertFound, alertFoundErr := s.alertDatastore.GetAlert(ctx, testAlert.GetId())
+	s.NoError(alertFoundErr)
+	s.True(alertFound)
+	s.Equal(storage.ViolationState_RESOLVED, alert.GetState())
+	_, secretFound, secretGetErr := s.secretDatastore.GetSecret(ctx, testSecret.GetId())
+	s.NoError(secretGetErr)
+	s.False(secretFound)
+	_, serviceAccountFound, serviceAccountGetErr := s.serviceAccountDatastore.GetServiceAccount(ctx, testServiceAccount.GetId())
+	s.NoError(serviceAccountGetErr)
+	s.False(serviceAccountFound)
+	_, roleFound, roleGetErr := s.roleDatastore.GetRole(ctx, testRole.GetId())
+	s.NoError(roleGetErr)
+	s.False(roleFound)
+	_, roleBindingFound, roleBindingGetErr := s.roleBindingDatastore.GetRoleBinding(ctx, testRole.GetId())
+	s.NoError(roleBindingGetErr)
+	s.False(roleBindingFound)
+	_, imageIntegrationFound, imageIntegrationGetErr := s.imageIntegrationDatastore.GetImageIntegration(ctx, testImageIntegration.GetId())
+	s.NoError(imageIntegrationGetErr)
+	s.False(imageIntegrationFound)
+}
+
+func (s *ClusterPostgresDataStoreTestSuite) TestPopulateClusterHealthInfo() {
+	ctx := sac.WithAllAccess(context.Background())
+
+	t := time.Now()
+	ts := protoconv.ConvertTimeToTimestamp(t)
+	idSuffixes := []string{"1", "2", "3"}
+	existingHealths := []*storage.ClusterHealthStatus{
+		{
+			SensorHealthStatus: storage.ClusterHealthStatus_HEALTHY,
+			LastContact:        ts,
+		},
+		{
+			SensorHealthStatus: storage.ClusterHealthStatus_DEGRADED,
+			LastContact:        ts,
+		},
+		{
+			SensorHealthStatus: storage.ClusterHealthStatus_UNHEALTHY,
+			LastContact:        ts,
+		},
+	}
+	for i, idSuffix := range idSuffixes {
+		cluster := &storage.Cluster{Name: "Cluster" + idSuffix, MainImage: mainImage, CentralApiEndpoint: centralEndpoint}
+		clusterId, err := s.clusterDatastore.AddCluster(ctx, cluster)
+		s.NoError(err)
+		s.NotEmpty(clusterId)
+		cluster.Id = clusterId
+		err = s.clusterDatastore.UpdateClusterHealth(ctx, clusterId, existingHealths[i])
+		s.NoError(err)
+	}
+	results, err := s.clusterDatastore.SearchRawClusters(ctx, pkgSearch.EmptyQuery())
+	s.NoError(err)
+	for i, result := range results {
```

Comment:
I'm not sure there should be any guarantee that the database will return the clusters in the order they were added. In order to avoid flakes, it might make sense to map the health objects by expected cluster name.